include/swift/Parse/Parser.h

```diff
@@ -956,7 +956,7 @@ class Parser {
   bool parseDifferentiableAttributeArguments(
       SmallVectorImpl<ParsedAutoDiffParameter> &params,
       Optional<DeclNameWithLoc> &jvpSpec, Optional<DeclNameWithLoc> &vjpSpec,
-      TrailingWhereClause *&whereClause);
+      bool &linear, TrailingWhereClause *&whereClause);
```

Comment:
These arguments should be ordered as they are presented in the syntax. `bool &linear` should be the first parameter.